## Librerías

In [92]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from scipy import stats
import re


## Carga de datos

In [80]:
df= pd.read_parquet('/Users/isaromobru/Desktop/DS102024_/ordenadores/train.parquet')

In [72]:
df
df.copy()

laptop_ID  Company                                  Product  \
0          599     Dell                            Inspiron 3179   
1           92     Asus  FX550IK-DM018T (FX-9830P/8GB/1TB/Radeon   
2          744       HP                                 ZBook 15   
3          729       HP                            Pavilion X360   
4          957  Toshiba                          Tecra A50-C-1ZV   
..         ...      ...                                      ...   
907        837     Acer                            Chromebook 11   
908        626     Acer     CB5-132T-C9KK (N3160/4GB/32GB/Chrome   
909        225     Dell                            Inspiron 3576   
910         17     Dell                            Inspiron 3567   
911         87     Dell                            Inspiron 7577   

               TypeName Inches                ScreenResolution  \
0    2 in 1 Convertible   11.6            Touchscreen 1366x768   
1                Gaming   15.6               Full HD 1920x1080   
2           Workstation   15.6               Full HD 1920x1080   
3    2 in 1 Convertible   13.3  IPS Panel Touchscreen 1366x768   
4              Notebook   15.6                        1366x768   
..                  ...    ...                             ...   
907             Netbook   11.6              IPS Panel 1366x768   
908  2 in 1 Convertible   11.6  IPS Panel Touchscreen 1366x768   
909            Notebook   15.6               Full HD 1920x1080   
910            Notebook   15.6               Full HD 1920x1080   
911              Gaming   15.6     IPS Panel Full HD 1920x1080   

                                      Cpu   Ram                Memory  \
0                Intel Core M 7Y30 1.0GHz   4GB             128GB SSD   
1                       AMD FX 9830P 3GHz   8GB               1TB HDD   
2             Intel Core i7 7700HQ 2.8GHz  16GB             256GB SSD   
3              Intel Core i5 7200U 2.5GHz   8GB             256GB SSD   
4              Intel Core i5 6200U 2.3GHz   8GB             256GB SSD   
..                                    ...   ...                   ...   
907  Intel Celeron Dual Core N3060 1.6GHz   4GB    32GB Flash Storage   
908  Intel Celeron Quad Core N3160 1.6GHz   4GB    32GB Flash Storage   
909            Intel Core i5 8250U 1.6GHz   8GB             256GB SSD   
910            Intel Core i7 7500U 2.7GHz   8GB             256GB SSD   
911           Intel Core i7 7700HQ 2.8GHz  16GB  128GB SSD +  1TB HDD   

                            Gpu       OpSys  Weight  Price_euros  
0         Intel HD Graphics 615  Windows 10  1.39kg       603.00  
1             AMD Radeon RX 560  Windows 10  2.45kg       699.00  
2           Nvidia Quadro M2200  Windows 10   2.6kg      2419.00  
3         Intel HD Graphics 620  Windows 10  1.58kg       836.60  
4         Intel HD Graphics 520  Windows 10   2.2kg      1119.00  
..                          ...         ...     ...          ...  
907       Intel HD Graphics 400   Chrome OS  1.35kg       355.00  
908       Intel HD Graphics 400   Chrome OS  1.25kg       379.00  
909              AMD Radeon 520       Linux   2.2kg       677.35  
910          AMD Radeon R5 M430  Windows 10   2.2kg       745.00  
911  Nvidia GeForce GTX 1050 Ti  Windows 10  2.65kg      1349.00  

[912 rows x 13 columns]

In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 912 entries, 0 to 911
Data columns (total 13 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   laptop_ID         912 non-null    int64  
 1   Company           912 non-null    object 
 2   Product           912 non-null    object 
 3   TypeName          912 non-null    object 
 4   Inches            912 non-null    object 
 5   ScreenResolution  912 non-null    object 
 6   Cpu               912 non-null    object 
 7   Ram               912 non-null    object 
 8   Memory            912 non-null    object 
 9   Gpu               912 non-null    object 
 10  OpSys             912 non-null    object 
 11  Weight            912 non-null    object 
 12  Price_euros       912 non-null    float64
dtypes: float64(1), int64(1), object(11)
memory usage: 92.8+ KB


In [8]:
df.shape

(912, 13)

In [9]:
df.columns

Index(['laptop_ID', 'Company', 'Product', 'TypeName', 'Inches',
       'ScreenResolution', 'Cpu', 'Ram', 'Memory', 'Gpu', 'OpSys', 'Weight',
       'Price_euros'],
      dtype='object')

In [12]:
df.isnull().sum()

laptop_ID           0
Company             0
Product             0
TypeName            0
Inches              0
ScreenResolution    0
Cpu                 0
Ram                 0
Memory              0
Gpu                 0
OpSys               0
Weight              0
Price_euros         0
dtype: int64

In [14]:
df['Company'].value_counts()

Company
Dell         214
Lenovo       206
HP           184
Asus         105
Acer          72
MSI           44
Toshiba       35
Apple         13
Samsung        9
Razer          5
Mediacom       5
Google         3
LG             3
Microsoft      3
Fujitsu        3
Xiaomi         2
Huawei         2
Chuwi          2
Vero           2
Name: count, dtype: int64

In [15]:
df['Product'].value_counts()

Product
XPS 13            21
Inspiron 3567     18
Vostro 3568       14
Inspiron 5570     13
250 G6            13
                  ..
Pavilion Power     1
Thinkpad L560      1
ThinkPad P51s      1
250 G4             1
Leopard GP72M      1
Name: count, Length: 484, dtype: int64

In [16]:
df['TypeName'].value_counts()

TypeName
Notebook              498
Gaming                149
Ultrabook             132
2 in 1 Convertible     87
Workstation            24
Netbook                22
Name: count, dtype: int64

In [17]:
df['Inches'].value_counts()

Inches
15.6    461
14.0    131
17.3    118
13.3    117
12.5     28
11.6     25
12.3      5
13.9      5
12.0      5
15.0      4
10.1      3
13.5      3
13.0      2
15.4      1
18.4      1
11.3      1
14.1      1
17.0      1
Name: count, dtype: int64

In [18]:
df['Memory'].value_counts()

Memory
256GB SSD                        282
1TB HDD                          163
500GB HDD                         94
512GB SSD                         77
128GB SSD +  1TB HDD              69
128GB SSD                         56
256GB SSD +  1TB HDD              52
32GB Flash Storage                27
64GB Flash Storage                12
512GB SSD +  1TB HDD              12
2TB HDD                           10
1.0TB Hybrid                       9
1TB SSD                            9
16GB Flash Storage                 6
256GB SSD +  2TB HDD               5
256GB Flash Storage                5
32GB SSD                           3
180GB SSD                          2
512GB Flash Storage                2
512GB SSD +  2TB HDD               2
128GB Flash Storage                2
256GB SSD +  500GB HDD             1
512GB SSD +  256GB SSD             1
16GB SSD                           1
8GB SSD                            1
64GB Flash Storage +  1TB HDD      1
128GB SSD +  2TB HDD           

In [19]:
df['OpSys'].value_counts()

OpSys
Windows 10      760
Linux            43
No OS            39
Windows 7        31
Chrome OS        20
Mac OS X          7
macOS             6
Windows 10 S      5
Android           1
Name: count, dtype: int64

In [20]:
df['Price_euros'].value_counts()

Price_euros
1099.0    10
899.0     10
1799.0    10
499.0     10
1499.0    10
          ..
395.0      1
1272.0     1
599.9      1
355.0      1
445.9      1
Name: count, Length: 598, dtype: int64

In [21]:
df['Product'].value_counts()

Product
XPS 13            21
Inspiron 3567     18
Vostro 3568       14
Inspiron 5570     13
250 G6            13
                  ..
Pavilion Power     1
Thinkpad L560      1
ThinkPad P51s      1
250 G4             1
Leopard GP72M      1
Name: count, Length: 484, dtype: int64

In [22]:
df['Ram'].value_counts()

Ram
8GB     436
4GB     262
16GB    138
6GB      30
12GB     18
2GB      14
32GB     12
24GB      1
64GB      1
Name: count, dtype: int64

In [23]:
df['ScreenResolution'].value_counts()

ScreenResolution
Full HD 1920x1080                                357
1366x768                                         201
IPS Panel Full HD 1920x1080                      148
IPS Panel Full HD / Touchscreen 1920x1080         42
Full HD / Touchscreen 1920x1080                   37
1600x900                                          19
Touchscreen 1366x768                              12
Quad HD+ / Touchscreen 3200x1800                  11
IPS Panel 4K Ultra HD 3840x2160                    8
4K Ultra HD / Touchscreen 3840x2160                6
IPS Panel Retina Display 2304x1440                 5
4K Ultra HD 3840x2160                              5
IPS Panel 4K Ultra HD / Touchscreen 3840x2160      5
Touchscreen 2560x1440                              5
IPS Panel Retina Display 2560x1600                 4
IPS Panel 1366x768                                 4
IPS Panel 2560x1440                                4
IPS Panel Touchscreen 1366x768                     3
Touchscreen 2400x1600        

In [24]:
df['TypeName'].value_counts()

TypeName
Notebook              498
Gaming                149
Ultrabook             132
2 in 1 Convertible     87
Workstation            24
Netbook                22
Name: count, dtype: int64

In [25]:
df['Weight'].value_counts()

Weight
2.2kg     84
2.1kg     38
2.4kg     31
2.5kg     29
2.3kg     27
          ..
1.27kg     1
1.31kg     1
2.21kg     1
2.99kg     1
3.8kg      1
Name: count, Length: 160, dtype: int64

In [27]:
df['laptop_ID'].value_counts()

laptop_ID
87      1
599     1
92      1
744     1
729     1
       ..
1126    1
917     1
382     1
91      1
529     1
Name: count, Length: 912, dtype: int64

In [28]:
df['Cpu'].value_counts()

Cpu
Intel Core i5 7200U 2.5GHz              137
Intel Core i7 7700HQ 2.8GHz             112
Intel Core i7 7500U 2.7GHz               88
Intel Core i5 6200U 2.3GHz               53
Intel Core i5 8250U 1.6GHz               49
                                       ... 
AMD E-Series 9000 2.2GHz                  1
Intel Core i7 7700HQ 2.7GHz               1
Intel Core M m3-7Y30 2.2GHz               1
Intel Pentium Dual Core 4405Y 1.5GHz      1
Intel Atom X5-Z8350 1.44GHz               1
Name: count, Length: 104, dtype: int64

In [29]:
df['Gpu'].value_counts()

Gpu
Intel HD Graphics 620      198
Intel HD Graphics 520      130
Nvidia GeForce GTX 1050     50
Intel UHD Graphics 620      42
Nvidia GeForce GTX 1060     34
                          ... 
AMD FirePro W4190M           1
Intel Iris Graphics 540      1
AMD Radeon 540               1
Nvidia GeForce GTX 970M      1
Nvidia GeForce 920           1
Name: count, Length: 94, dtype: int64

### Cambiar tipo de dato

In [30]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 912 entries, 0 to 911
Data columns (total 13 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   laptop_ID         912 non-null    int64  
 1   Company           912 non-null    object 
 2   Product           912 non-null    object 
 3   TypeName          912 non-null    object 
 4   Inches            912 non-null    object 
 5   ScreenResolution  912 non-null    object 
 6   Cpu               912 non-null    object 
 7   Ram               912 non-null    object 
 8   Memory            912 non-null    object 
 9   Gpu               912 non-null    object 
 10  OpSys             912 non-null    object 
 11  Weight            912 non-null    object 
 12  Price_euros       912 non-null    float64
dtypes: float64(1), int64(1), object(11)
memory usage: 92.8+ KB


In [34]:
df

laptop_ID  Company                                  Product  \
0          599     Dell                            Inspiron 3179   
1           92     Asus  FX550IK-DM018T (FX-9830P/8GB/1TB/Radeon   
2          744       HP                                 ZBook 15   
3          729       HP                            Pavilion X360   
4          957  Toshiba                          Tecra A50-C-1ZV   
..         ...      ...                                      ...   
907        837     Acer                            Chromebook 11   
908        626     Acer     CB5-132T-C9KK (N3160/4GB/32GB/Chrome   
909        225     Dell                            Inspiron 3576   
910         17     Dell                            Inspiron 3567   
911         87     Dell                            Inspiron 7577   

               TypeName Inches                ScreenResolution  \
0    2 in 1 Convertible   11.6            Touchscreen 1366x768   
1                Gaming   15.6               Full HD 1920x1080   
2           Workstation   15.6               Full HD 1920x1080   
3    2 in 1 Convertible   13.3  IPS Panel Touchscreen 1366x768   
4              Notebook   15.6                        1366x768   
..                  ...    ...                             ...   
907             Netbook   11.6              IPS Panel 1366x768   
908  2 in 1 Convertible   11.6  IPS Panel Touchscreen 1366x768   
909            Notebook   15.6               Full HD 1920x1080   
910            Notebook   15.6               Full HD 1920x1080   
911              Gaming   15.6     IPS Panel Full HD 1920x1080   

                                      Cpu   Ram                Memory  \
0                Intel Core M 7Y30 1.0GHz   4GB             128GB SSD   
1                       AMD FX 9830P 3GHz   8GB               1TB HDD   
2             Intel Core i7 7700HQ 2.8GHz  16GB             256GB SSD   
3              Intel Core i5 7200U 2.5GHz   8GB             256GB SSD   
4              Intel Core i5 6200U 2.3GHz   8GB             256GB SSD   
..                                    ...   ...                   ...   
907  Intel Celeron Dual Core N3060 1.6GHz   4GB    32GB Flash Storage   
908  Intel Celeron Quad Core N3160 1.6GHz   4GB    32GB Flash Storage   
909            Intel Core i5 8250U 1.6GHz   8GB             256GB SSD   
910            Intel Core i7 7500U 2.7GHz   8GB             256GB SSD   
911           Intel Core i7 7700HQ 2.8GHz  16GB  128GB SSD +  1TB HDD   

                            Gpu       OpSys  Weight  Price_euros  
0         Intel HD Graphics 615  Windows 10  1.39kg       603.00  
1             AMD Radeon RX 560  Windows 10  2.45kg       699.00  
2           Nvidia Quadro M2200  Windows 10   2.6kg      2419.00  
3         Intel HD Graphics 620  Windows 10  1.58kg       836.60  
4         Intel HD Graphics 520  Windows 10   2.2kg      1119.00  
..                          ...         ...     ...          ...  
907       Intel HD Graphics 400   Chrome OS  1.35kg       355.00  
908       Intel HD Graphics 400   Chrome OS  1.25kg       379.00  
909              AMD Radeon 520       Linux   2.2kg       677.35  
910          AMD Radeon R5 M430  Windows 10   2.2kg       745.00  
911  Nvidia GeForce GTX 1050 Ti  Windows 10  2.65kg      1349.00  

[912 rows x 13 columns]

In [81]:
#inches
df['Inches']=df['Inches'].astype('float')

In [82]:
# Weight
# voy a comprobar si todo esta en kg 
# Verifica si TODOS los valores contienen 'kg'
cantidad_con_kg = df['Weight'].str.contains('kg', case=False).sum()
print(f"Valores con 'kg': {cantidad_con_kg} de {len(df)}")

# Cambiar la etiqueta e indicar que esta en kg 
df.rename(columns={'Weight': 'Weight_kg'}, inplace=True)
# borrar kg y pasarlo a numerico
df['Weight_kg'] = df['Weight_kg'].str.replace('kg', '', regex=False).str.strip()
df['Weight_kg'] = df['Weight_kg'].astype(float)

# Mostrar el resultado
print(df)

Valores con 'kg': 912 de 912
     laptop_ID  Company                                  Product  \
0          599     Dell                            Inspiron 3179   
1           92     Asus  FX550IK-DM018T (FX-9830P/8GB/1TB/Radeon   
2          744       HP                                 ZBook 15   
3          729       HP                            Pavilion X360   
4          957  Toshiba                          Tecra A50-C-1ZV   
..         ...      ...                                      ...   
907        837     Acer                            Chromebook 11   
908        626     Acer     CB5-132T-C9KK (N3160/4GB/32GB/Chrome   
909        225     Dell                            Inspiron 3576   
910         17     Dell                            Inspiron 3567   
911         87     Dell                            Inspiron 7577   

               TypeName  Inches                ScreenResolution  \
0    2 in 1 Convertible    11.6            Touchscreen 1366x768   
1                Gam

In [83]:
#ram
# Comprobar si todo esta GB
cantidad_con_gb = df['Ram'].str.contains('GB', case=False).sum()
print(f"Valores con 'GB': {cantidad_con_gb} de {len(df)}")
# Cambiar la etiqueta e indicar que esta en GB
df.rename(columns={'Ram': 'Ram_GB'}, inplace=True)
# borrar GB y pasarlo a numerico
df['Ram_GB'] = df['Ram_GB'].str.replace('GB', '', regex=False).str.strip()
df['Ram_GB'] = df['Ram_GB'].astype(int)

Valores con 'GB': 912 de 912


In [86]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 912 entries, 0 to 911
Data columns (total 13 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   laptop_ID         912 non-null    int64  
 1   Company           912 non-null    object 
 2   Product           912 non-null    object 
 3   TypeName          912 non-null    object 
 4   Inches            912 non-null    float64
 5   ScreenResolution  912 non-null    object 
 6   Cpu               912 non-null    object 
 7   Ram_GB            912 non-null    int64  
 8   Memory            912 non-null    object 
 9   Gpu               912 non-null    object 
 10  OpSys             912 non-null    object 
 11  Weight_kg         912 non-null    float64
 12  Price_euros       912 non-null    float64
dtypes: float64(3), int64(2), object(8)
memory usage: 92.8+ KB


Memory no se que hacer

In [89]:
# Convertir los object en categoricas 
text_columns = ['Company', 'Product', 'TypeName', 'ScreenResolution', 'Cpu', 
                'Gpu', 'OpSys']

for col in text_columns:
    df[col] = df[col].astype('category')

In [84]:
df

laptop_ID  Company                                  Product  \
0          599     Dell                            Inspiron 3179   
1           92     Asus  FX550IK-DM018T (FX-9830P/8GB/1TB/Radeon   
2          744       HP                                 ZBook 15   
3          729       HP                            Pavilion X360   
4          957  Toshiba                          Tecra A50-C-1ZV   
..         ...      ...                                      ...   
907        837     Acer                            Chromebook 11   
908        626     Acer     CB5-132T-C9KK (N3160/4GB/32GB/Chrome   
909        225     Dell                            Inspiron 3576   
910         17     Dell                            Inspiron 3567   
911         87     Dell                            Inspiron 7577   

               TypeName  Inches                ScreenResolution  \
0    2 in 1 Convertible    11.6            Touchscreen 1366x768   
1                Gaming    15.6               Full HD 1920x1080   
2           Workstation    15.6               Full HD 1920x1080   
3    2 in 1 Convertible    13.3  IPS Panel Touchscreen 1366x768   
4              Notebook    15.6                        1366x768   
..                  ...     ...                             ...   
907             Netbook    11.6              IPS Panel 1366x768   
908  2 in 1 Convertible    11.6  IPS Panel Touchscreen 1366x768   
909            Notebook    15.6               Full HD 1920x1080   
910            Notebook    15.6               Full HD 1920x1080   
911              Gaming    15.6     IPS Panel Full HD 1920x1080   

                                      Cpu  Ram_GB                Memory  \
0                Intel Core M 7Y30 1.0GHz       4             128GB SSD   
1                       AMD FX 9830P 3GHz       8               1TB HDD   
2             Intel Core i7 7700HQ 2.8GHz      16             256GB SSD   
3              Intel Core i5 7200U 2.5GHz       8             256GB SSD   
4              Intel Core i5 6200U 2.3GHz       8             256GB SSD   
..                                    ...     ...                   ...   
907  Intel Celeron Dual Core N3060 1.6GHz       4    32GB Flash Storage   
908  Intel Celeron Quad Core N3160 1.6GHz       4    32GB Flash Storage   
909            Intel Core i5 8250U 1.6GHz       8             256GB SSD   
910            Intel Core i7 7500U 2.7GHz       8             256GB SSD   
911           Intel Core i7 7700HQ 2.8GHz      16  128GB SSD +  1TB HDD   

                            Gpu       OpSys  Weight_kg  Price_euros  
0         Intel HD Graphics 615  Windows 10       1.39       603.00  
1             AMD Radeon RX 560  Windows 10       2.45       699.00  
2           Nvidia Quadro M2200  Windows 10       2.60      2419.00  
3         Intel HD Graphics 620  Windows 10       1.58       836.60  
4         Intel HD Graphics 520  Windows 10       2.20      1119.00  
..                          ...         ...        ...          ...  
907       Intel HD Graphics 400   Chrome OS       1.35       355.00  
908       Intel HD Graphics 400   Chrome OS       1.25       379.00  
909              AMD Radeon 520       Linux       2.20       677.35  
910          AMD Radeon R5 M430  Windows 10       2.20       745.00  
911  Nvidia GeForce GTX 1050 Ti  Windows 10       2.65      1349.00  

[912 rows x 13 columns]

In [90]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 912 entries, 0 to 911
Data columns (total 13 columns):
 #   Column            Non-Null Count  Dtype   
---  ------            --------------  -----   
 0   laptop_ID         912 non-null    int64   
 1   Company           912 non-null    category
 2   Product           912 non-null    category
 3   TypeName          912 non-null    category
 4   Inches            912 non-null    float64 
 5   ScreenResolution  912 non-null    category
 6   Cpu               912 non-null    category
 7   Ram_GB            912 non-null    int64   
 8   Memory            912 non-null    object  
 9   Gpu               912 non-null    category
 10  OpSys             912 non-null    category
 11  Weight_kg         912 non-null    float64 
 12  Price_euros       912 non-null    float64 
dtypes: category(7), float64(3), int64(2), object(1)
memory usage: 80.2+ KB


In [103]:
#ScreenResolution
#Agrupar las medidas en una nueva columna 
df['Resolución'] = df['ScreenResolution'].apply(
    lambda x: re.search(r'(\d{3,4}x\d{3,4})', x).group(0) if re.search(r'(\d{3,4}x\d{3,4})', x) else None)
#Crear otra variable para características de la pantalla



In [104]:
# Extraer y crear nuevas columnas
#df[['ancho', 'alto']] = df['ScreenResolution'].str.extract(r'(\d+)x(\d+)').astype(int)
df['tipo'] = df['ScreenResolution'].str.extract(r'^(\w+)')

In [105]:
df['tipo'].value_counts()

tipo
Full           394
IPS            241
1366x768       201
Touchscreen     26
1600x900        19
Quad            13
4K              11
1920x1080        3
1440x900         3
2560x1440        1
Name: count, dtype: int64

In [106]:
df

laptop_ID  Company                                  Product  \
0          599     Dell                            Inspiron 3179   
1           92     Asus  FX550IK-DM018T (FX-9830P/8GB/1TB/Radeon   
2          744       HP                                 ZBook 15   
3          729       HP                            Pavilion X360   
4          957  Toshiba                          Tecra A50-C-1ZV   
..         ...      ...                                      ...   
907        837     Acer                            Chromebook 11   
908        626     Acer     CB5-132T-C9KK (N3160/4GB/32GB/Chrome   
909        225     Dell                            Inspiron 3576   
910         17     Dell                            Inspiron 3567   
911         87     Dell                            Inspiron 7577   

               TypeName  Inches                ScreenResolution  \
0    2 in 1 Convertible    11.6            Touchscreen 1366x768   
1                Gaming    15.6               Full HD 1920x1080   
2           Workstation    15.6               Full HD 1920x1080   
3    2 in 1 Convertible    13.3  IPS Panel Touchscreen 1366x768   
4              Notebook    15.6                        1366x768   
..                  ...     ...                             ...   
907             Netbook    11.6              IPS Panel 1366x768   
908  2 in 1 Convertible    11.6  IPS Panel Touchscreen 1366x768   
909            Notebook    15.6               Full HD 1920x1080   
910            Notebook    15.6               Full HD 1920x1080   
911              Gaming    15.6     IPS Panel Full HD 1920x1080   

                                      Cpu  Ram_GB                Memory  \
0                Intel Core M 7Y30 1.0GHz       4             128GB SSD   
1                       AMD FX 9830P 3GHz       8               1TB HDD   
2             Intel Core i7 7700HQ 2.8GHz      16             256GB SSD   
3              Intel Core i5 7200U 2.5GHz       8             256GB SSD   
4              Intel Core i5 6200U 2.3GHz       8             256GB SSD   
..                                    ...     ...                   ...   
907  Intel Celeron Dual Core N3060 1.6GHz       4    32GB Flash Storage   
908  Intel Celeron Quad Core N3160 1.6GHz       4    32GB Flash Storage   
909            Intel Core i5 8250U 1.6GHz       8             256GB SSD   
910            Intel Core i7 7500U 2.7GHz       8             256GB SSD   
911           Intel Core i7 7700HQ 2.8GHz      16  128GB SSD +  1TB HDD   

                            Gpu       OpSys  Weight_kg  Price_euros  \
0         Intel HD Graphics 615  Windows 10       1.39       603.00   
1             AMD Radeon RX 560  Windows 10       2.45       699.00   
2           Nvidia Quadro M2200  Windows 10       2.60      2419.00   
3         Intel HD Graphics 620  Windows 10       1.58       836.60   
4         Intel HD Graphics 520  Windows 10       2.20      1119.00   
..                          ...         ...        ...          ...   
907       Intel HD Graphics 400   Chrome OS       1.35       355.00   
908       Intel HD Graphics 400   Chrome OS       1.25       379.00   
909              AMD Radeon 520       Linux       2.20       677.35   
910          AMD Radeon R5 M430  Windows 10       2.20       745.00   
911  Nvidia GeForce GTX 1050 Ti  Windows 10       2.65      1349.00   

    Resolución  ancho  alto         tipo  
0     1366x768   1366   768  Touchscreen  
1    1920x1080   1920  1080         Full  
2    1920x1080   1920  1080         Full  
3     1366x768   1366   768          IPS  
4     1366x768   1366   768     1366x768  
..         ...    ...   ...          ...  
907   1366x768   1366   768          IPS  
908   1366x768   1366   768          IPS  
909  1920x1080   1920  1080         Full  
910  1920x1080   1920  1080         Full  
911  1920x1080   1920  1080          IPS  

[912 rows x 17 columns]

Inches:✅
Verifica que los valores sean numéricos. Convierte a tipo float si no lo están.
Busca inconsistencias (valores extraños como símbolos).

ScreenResolution:
Divide esta columna en subcolumnas:
Resolución (e.g., "1920x1080").
Características adicionales (e.g., "Touchscreen", "IPS Panel").
Asegúrate de que las resoluciones estén en el formato estándar (ancho x alto).


Cpu:
Extrae el modelo de procesador (e.g., "Core i7").
Divide en subcolumnas como:
Marca (e.g., "Intel", "AMD").
Modelo (e.g., "Core i7").
Velocidad (e.g., "2.8GHz").


Ram:✅
Asegúrate de que todas las entradas terminen en GB.
Convierte los valores en enteros para representar solo el número (e.g., "8GB" → 8).


Memory:
Maneja combinaciones de almacenamiento (e.g., "256GB SSD + 1TB HDD").
Crea subcolumnas como:
Tipo de almacenamiento (SSD, HDD, Flash Storage).
Capacidad (convierte a una unidad estándar como GB).


Gpu:
Extrae la marca y el modelo de GPU (e.g., "Intel HD Graphics 620" → Marca: Intel, Modelo: HD Graphics 620).


OpSys:
Normaliza nombres de sistemas operativos (e.g., "Windows 10" → "Windows").
Identifica valores faltantes o no válidos.


Weight:✅
Asegúrate de que los valores sean numéricos (remueve "kg").
Convierte a tipo float.


Price_euros:✅
Verifica que sea numérico. Convierte a tipo float si no lo está.
